roommanager.js

```diff
@@ -35,24 +35,20 @@ module.exports = function (server, storage) {
 	}
 
 	function updateRoom(room) {
-		if (!_.isEmpty(room.currentSource) && room.currentSource.hasOwnProperty("length")) {
-			// FIXME: the client expectes playbackDuration to exist and be >0.
-			// A better way to do this would be for the client to infer playbackDuration
-			// from currentsource.length
-			room.playbackDuration = room.currentSource.length;
-		}
+		
+			
 
 		if (_.isEmpty(room.currentSource) && room.queue.length > 0) {
 			room.currentSource = room.queue.shift();
 		}
-		else if (!_.isEmpty(room.currentSource) && room.playbackPosition > room.playbackDuration) {
+		else if (!_.isEmpty(room.currentSource) && room.playbackPosition > room.currentSource.length) {
 			room.currentSource = room.queue.length > 0 ? room.queue.shift() : {};
 			room.playbackPosition = 0;
 		}
 		if (_.isEmpty(room.currentSource) && room.queue.length == 0 && room.isPlaying) {
 			room.isPlaying = false;
 			room.playbackPosition = 0;
-			room.playbackDuration = 0;
+			room.currentSource.length = 0;
```

Comment:
This doesn't need to be set because it will be overwritten anyway by the next video.